### PR TITLE
Update to new build tools and gradle version

### PR DESCRIPTION
**Pull request by PGrube26**
Hi,
thanks for this library.
We need to use the newest gradle version and support libraries so I updated your library. Furthermore I had to implement a small fix for a crash coming with the new support library 25.3.0 where fragments without host can not create a new ChildFragmentManager. I just added a check for the host/activity and now it works again.

### CODE REVIEW

File: demo/build.gradle [hunk 1]
```diff
@@ -23,8 +23,8 @@ android {
     lintOptions {
         abortOnError true
 
-        htmlOutput file("$project.buildDir/reports/lint/lint.html")
-        xmlOutput file("$project.buildDir/reports/lint/lint.xml")
+        htmlOutput file("$project.buildDir/reports/lint/android_task_lint.html")
```
Comment (vRallev) [inline demo/build.gradle]: Feel free to remove both lines completely

File: library/build.gradle [hunk 2]
```diff
@@ -27,8 +27,8 @@ android {
     lintOptions {
         abortOnError true
 
-        htmlOutput file("$project.buildDir/reports/lint/lint.html")
-        xmlOutput file("$project.buildDir/reports/lint/lint.xml")
+        htmlOutput file("$project.buildDir/reports/lint/android_task_lint.html")
```
Comment (vRallev) [inline library/build.gradle]: Remove both lines

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Wed Apr 27 14:30:01 CEST 2016
+#Thu Mar 16 16:54:40 CET 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.13-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.4.1-all.zip
```
Comment (vRallev) [inline gradle/wrapper/gradle-wrapper.properties:6]: You didn't push the gradle .jar files. The gradle wrapper should be updated with `./gradlew wrapper --gradle-version 3.4.1 --distribution-type all`

**Review by vRallev — CHANGES_REQUESTED** [review]
A few minor comments. Only problem is that the wrapper should be updated properly. Rest looks good.

**Comment by PGrube26** [comment]
Ok, now with custom lint output path removed and gradlew updated as you said.

**Comment by vRallev** [comment]
Thanks!

**Pull request merged**